core/modules/locale/locale.module

```diff
@@ -590,10 +600,11 @@ function locale($string = NULL, $context = NULL, $langcode = NULL) {
   if (!isset($locale_t[$langcode][$context][$string])) {
     // We do not have this translation cached, so get it from the DB.
     try {
-      $translation = db_query("SELECT s.lid, t.translation, s.version FROM {locales_source} s LEFT JOIN {locales_target} t ON s.lid = t.lid AND t.language = :language WHERE s.source = :source AND s.context = :context", array(
+      $translation = db_query("SELECT s.lid, t.translation, s.version FROM {locales_source} s LEFT JOIN {locales_target} t ON s.lid = t.lid AND t.language = :language WHERE s.source = :source AND s.context = :context AND s.textgroup = 'default'", array(
         ':language' => $langcode,
         ':source' => $string,
         ':context' => (string) $context,
+        'textgroup' => 'default',
```

Comment:
Besides the fact that it probably should be ":textgroup" (with colon), you're not using that placeholder in your query.